src/app/charts/charts.component.ts

```diff
@@ -31,38 +40,56 @@ export class ChartsComponent implements OnInit {
     },
   ];
 
-  constructor() {}
+  constructor(private route: ActivatedRoute) {}
 
   ngOnInit(): void {
     this.lineChartData = this.labels.reduce(
       (acc, el) => [...acc, { data: [], label: el }],
       []
     );
+
     this.loadChartData(this.labels[0]);
   }
 
-  onClick(type: string) {
+  ngOnChanges(changes: SimpleChanges) {
```

Comment:
Could this be onInit? If the route is changed this component should be reinitialized, right?

Also, you don't need id variable.